### ISSUE TITLE
[Error] Command 'Markdown Links: Show Graph' resulted in an error ([UriError]: Scheme contains illegal characters.)

### ISSUE THREAD

**Issue by Annegrim** [tchayen/markdown-links#45]
Since two days ago I get an error when commanding `Markdown Links: Show Graph`:

> Command 'Markdown Links: Show Graph' resulted in an error ([UriError]: Scheme contains illegal characters.)

I use the `"markdown-links.fileIdRegexp": "(?<=^#).+$",` setting which I got from [here](https://github.com/tchayen/markdown-links/blob/master/docs/id-based-links.md). My folder has ~1.500 files

**Comment by felixbinder** [comment]
I get the same error since a few days ago as well, without anything having changed and the default value for fileIdRegexp.

**Comment by felixbinder** [comment]
Downgrading the extension to 0.60 makes the graph work again.

**Comment by Annegrim** [comment]
Is there a specific trick to it? I reverted to 0.60 and now I don't even see the `Markdown Links: Show Graph` command. Even restarting the PC didn't do the trick.

**Comment by felixbinder** [comment]
No, I simply right-clicked the extension and installed 0.60. Perhaps you could try to disable and enable it?

**Comment by Annegrim** [comment]
That didn't work for me.

**Comment by tchayen** [comment]
I believe that below 0.6.0 the command was simply `Show Graph` (changed to stick to the convention). That might answer why it disappeared for you.

Regarding the scheme problem – I am not sure if it is related to the file ID regexp, but i have no better idea for now how to check it.

**Comment by Am3ra** [comment]
I had the same error, but it was fixed by checking my links. One of my web links was badly mangled, and caused that error. I recommend checking that all your links are correctly formatted.

**Comment by nrgapple** [comment]
Yep, downgrading to 0.6.0 worked for me also.

**Comment by tchayen** [comment]
Ok, so two things would help:
- Checking what has changed since `0.6.0` in the way parsing happens.
- Providing better error handling that wouldn't let a parsing error blow the whole graph.

**Comment by nrgapple** [comment]
@tchayen yes exactly. If the error would point to the file and line in the files that is causing the parsing error that would be extremely helpful

**Comment by petterobam** [comment]
message to watch, when fixed then upgrade

**Comment by RyanPersson** [comment]
I had this same error occur in v0.80 . 
```
Command 'Markdown Links: Show Graph' resulted in an error ([UriError]: Scheme contains illegal characters.)
```


I figured it was due to a character sequence in one of my files so I worked through all the changes since my last commit to my notes.
I was able to trace it down to this line added in a markdown file:

`[quanta magazine]((https://www.quantamagazine.org/machine-learnings-amazing-ability-to-predict-chaos-20180418/))`

Changing the above line to

`[quanta magazine](https://www.quantamagazine.org/machine-learnings-amazing-ability-to-predict-chaos-20180418/)`

fixes the issue & lets markdown links load normally.


I thought at first that it was because of the double parenthesis. But making a test file with `[hello]((world))` in a markdown file does not trigger the issue. So then I thought that it must be because of something in URL syntax, & sure enough
 `[test]((https://google.com))` reliably triggered it.
 I played around with it, and narrowed it down to the colon. 
`[]((:))` triggers the issue consistently for me, as well as any any string of the form, `[text]((text:text))`.


As a temporary fix for anyone having the same issue, try searching for `]((` and maybe `))` to find any links in your markdown files with accidental second parentheses containing a `:`.  Fixing these should resolve the issue.

**Comment by tchayen** [comment]
Wow, that one is strange. I will try looking for similar issues in `remark` as this is the underlying Markdown parser and they should have faced that too.

**Comment by patchworquill** [comment]
Just had this issue, searched for ages, then found @RyanPersson's solution: 

Used VSCode's _search_ for ```]((``` and that found an instance of this glitch! Removed the extra brackets and BOOM! graph working again. 

Thanks for the help.

**Comment by sadespresso** [comment]
I experienced the same issue on v0.8.0, and downgrading to 0.7.0 somehow fixed it...

I have a header with an emoji (✨), that might have caused it. Everything else is valid, escaped URLs (including `https://` links).

**Comment by AlexisBogroff** [comment]
Same and found the issue in my case: **a link contained a double parenthesis**

Unfortunately I couldn't find it by looking for "((" as specified above in this discussion, since it was "(<(".

My project contained node_module.
node_module < rollup < CHANGELOG.md < ```"[#675](<(https://github.com/rollup/rollup/issues/675)>)"```

The issue was solved by changing the link into: ```"[#675](<https://github.com/rollup/rollup/issues/675>)"```
i.e. removing the interior parenthesis.

The general misleading pattern was:
Issue:      ```[text](<(link)>)```
Solution: ```[text](<link>)```

**Comment by tchayen** [comment]
Fixed in https://github.com/tchayen/markdown-links/commit/60ca2fbb61a847bc22d45323a759107a090ce267
